### PR TITLE
Updating the documentation and keeping it up to date

### PR DESCRIPTION
**Pull request by ali-ramadhan**
In this PR I'll be updating the documentation, especially the model setup section, to reflect the many recent changes and additions to the API.

To avoid this problem of documentation going out of date, I will also be converting and adding `jldoctest`s so we should actually get an error when documentation goes out of date. A lot of `@example` blocks don't run anymore and we even have a `jldoctest` failure but Travis CI seems to pass so I'll look into making Travis CI fail the doc build if a doctest fails.

If we have a lot of doctests it might actually be easy to keep them updated: https://juliadocs.github.io/Documenter.jl/latest/man/doctests/#Fixing-Outdated-Doctests

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
I think this PR is ready for review now.

The model setup docs have been updated and most snippets have been converted to `jldoctest`s so we can ensure they stay up to date. I would like to expand the output writers section but I think we'll be modifying the API a bit there so might not be worth expanding just yet.

I also updated a few `Base.show` methods to make the output of some code snippets prettier.

Another big change is that if a doctest fails or goes out of date, the Travis CI documentation build will fail giving us a chance to fix and update the documentation before merging a PR. This is done by passing `strict=true` to the `makedocs` function.

Ideally we would use `makedocs(modules=[Oceananigans], checkdocs=:exports, strict=true, ...)` to check that every exported name has a docstring. But currently we have 40~50 missing docstrings so I disabled it (`checkdocs=:none`) until we can take care of these docstrings in a future PR.

File: docs/src/model_setup/coriolis.md [hunk 1]
```diff
@@ -7,58 +8,73 @@ three options are available: no rotation, $f$-plane, and $\beta$-plane.
...
 
 To set up an $f$-plane with non-traditional Coriolis terms, for example, with 
 $\bm{f} = (0, f_y, f_z) = (0, 2, 1) \times 10^{-4} \text{s}^{-1}$,
-```@example
-using Oceananigans # hide
-coriolis = NonTraditionalFPlane(fz=1e-4, fy=2e-4)
+
+```jldoctest
+julia> coriolis = NonTraditionalFPlane(fz=1e-4, fy=2e-4)
+Non-TraditionalFPlane{Float64}:fz = 1.00e-04, fy = 2.00e-04
```
Comment (glwagner) [inline docs/src/model_setup/coriolis.md:46]: 👏🏼

though I should say, missing a space after the colon

File: src/Coriolis/f_plane.jl [hunk 2]
```diff
@@ -95,4 +95,4 @@ end
 @inline z_f_cross_U(i, j, k, grid, coriolis::NonTraditionalFPlane, U) = - coriolis.fy * ℑxzᶜᵃᶠ(i, j, k, grid, U.u)
 
 Base.show(io::IO, f_plane::NonTraditionalFPlane{FT}) where FT =
-    println(io, "Non-Traditional FPlane{$FT}:", @sprintf("fz = %.2e, fy = %.2e", f_plane.fz, f_plane.fy))
+    println(io, "Non-TraditionalFPlane{$FT}:", @sprintf("fz = %.2e, fy = %.2e", f_plane.fz, f_plane.fy))
```
Comment (glwagner) [inline src/Coriolis/f_plane.jl:98]: space after colon?

File: src/Grids/grid_utils.jl [hunk 3]
```diff
@@ -102,19 +102,21 @@ on `grid` in the z-direction. For `Bounded` directions,
...
 
 julia> zC = znodes(Cell, horz_periodic_grid)
 3-element view(OffsetArray(::StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}, 0:4), 1:3) with eltype Float64:
  -0.8333333333333331
  -0.4999999999999999
  -0.16666666666666652
+```
 
- julia> zF = znodes(Face, horz_periodic_grid)
+``` jldoctest znodes
```
Comment (glwagner) [inline src/Grids/grid_utils.jl:118]: 👏🏼

File: docs/src/model_setup/coriolis.md (hunk 1 above)
Comment (ali-ramadhan) [reply docs/src/model_setup/coriolis.md]: Ah thanks for catching that!

**Pull request merged**